canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,215 @@
+
+            final ExecutorService executorService = Executors.newSingleThreadExecutor();
+            final GetMediaWorker getMediaWorker = GetMediaWorker.create(
+                    Regions.fromName(mRegion),
+                    mCredentialsProvider,
+                    mStreamName,
+                    startSelector,
+                    mAmazonKinesisVideo,
+                    frameVisitor);
+
+            final Future<?> task = executorService.submit(getMediaWorker);
+            task.get();
+            executorService.shutdown();
+
+        } catch (Exception e) {
+            log.error("Failed while calculating time to first fragment, {}", e);
+        }
+    }
+
+    protected static void publishMetricToCW(String metricName, double value, StandardUnit cwUnit) {
+        try {
+            log.info("Emitting the following metric: {} - {}", metricName, value);
+            final Dimension dimensionPerStream = new Dimension()
+                    .withName("StorageWebRTCSDKCanaryStreamName")
+                    .withValue(mStreamName);
+            final Dimension aggregatedDimension = new Dimension()
+                    .withName("StorageWebRTCSDKCanaryLabel")
+                    .withValue(mCanaryLabel);
+            List<MetricDatum> datumList = new ArrayList<>();
+
+            MetricDatum datum = new MetricDatum()
+                    .withMetricName(metricName)
+                    .withUnit(cwUnit)
+                    .withValue(value)
+                    .withDimensions(dimensionPerStream);
+            datumList.add(datum);
+            MetricDatum aggDatum = new MetricDatum()
+                    .withMetricName(metricName)
+                    .withUnit(cwUnit)
+                    .withValue(value)
+                    .withDimensions(aggregatedDimension);
+            datumList.add(aggDatum);
+
+            PutMetricDataRequest request = new PutMetricDataRequest()
+                    .withNamespace("KinesisVideoSDKCanary")
+                    .withMetricData(datumList);
+            mCwClient.putMetricData(request);
+        } catch (Exception e) {
+            log.error("Failed while while publishing metric to CW, {}", e);
+        }
+    }
+
+    public static void main(final String[] args) throws Exception {
+        final Integer canaryRunTime = Integer.parseInt(System.getenv("CANARY_DURATION_IN_SECONDS"));
+
+        log.info("Stream name: {}", mStreamName);
+
+        mCredentialsProvider = new EnvironmentVariableCredentialsProvider();
+        mAmazonKinesisVideo = AmazonKinesisVideoClientBuilder.standard()
+                .withRegion(mRegion)
+                .withCredentials(mCredentialsProvider)
+                .build();
+        mCwClient = AmazonCloudWatchClientBuilder.standard()
+                .withRegion(mRegion)
+                .withCredentials(mCredentialsProvider)
+                .build();
+
+        switch (mCanaryLabel) {
+            case "WebrtcLongRunning": {
```

Comment:
All these should be defined in a Constants.java class and used. It is not a good programming practice to hard code strings, because it makes it cumbersome to update all the locations it could be used in. Something like:

```
public final class Constants {

    // Example constants
    public static final String WEBRTC_LONG_RUNNING_LABEL = "WebrtcLongRunning";
    public static final STRING INTERVAL_TIMER_NAME = "IntervalMetricsTimer";

    // More constants can be added here

    private Constants() {
    }
}
```
  When you use it here, it should be:
  
  ```
  case Constants.WEBRTC_LONG_RUNNING_LABEL:
  ```